### PR TITLE
harden-ssh-k8s-api-ports

### PR DESCRIPTION
**Pull request by enekui**
## Description 
Added a set of new variables that will default to true to control SSH and Kubernetes API access to the EC2 instance.
Added a new Data source to conditionally get the public IP of the Terraform module operator. We are manipulating the output to get the clean public IP and pass it as the default allowed CIDR block to the SSH and Kubernetes API port. Also, there are two new variables to control this just in case it's needed to open SSH or Kubernetes API ports independently to everyone or other IPs/CIDR blocks. Removed unrelated left variables from ALB.
Updated README.md
Tested the new changes.

Note: *This is a small upgrade, but will make a huge difference in how customers take this from a security best practices point of view.*

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -1163,6 +1163,51 @@ module "ec2_sg" {
...
+      rule        = "kubernetes-api-tcp"
+      cidr_blocks = join(",", concat(local.local_public_ip, var.ec2_instance_kubernetes_api_cidr_blocks))
+    },
+  ]
+}
+
+# Data source to get the local public IP.
+data "http" "local_public_ip" {
+  count = var.enable_ec2_instance_ssh_access || var.enable_ec2_instance_kubernetes_api_access ? 1 : 0
+  url   = "http://ipv4.icanhazip.com"
```
Comment (bebosudo) [inline main.tf:1203]:
```suggestion
  url   = "https://ipv4.icanhazip.com"
```

File: variables.tf [hunk 2]
```diff
@@ -1643,12 +1643,44 @@ variable "ec2_instance_sg_egress_cidr_blocks" {
...
+  description = "Determines whether Kubernetes API access is enabled for the EC2 instance."
+  default     = true
+}
+
+## EC2 SSH allowed CIDR blocks
+variable "ec2_instnace_ssh_cidr_blocks" {
+  type        = list(string)
+  description = "The CIDR blocks for the security group ingress rule."
+  default     = [""]
+
```
Comment (bebosudo) [inline variables.tf:1672]:
```suggestion
```

File: variables.tf [hunk 3]
```diff
@@ -1643,12 +1643,44 @@ variable "ec2_instance_sg_egress_cidr_blocks" {
...
+}
+
+## EC2 instance Kubernetes API allowed CIDR blocks
+variable "ec2_instance_kubernetes_api_cidr_blocks" {
+  type        = list(string)
+  description = "The CIDR blocks for the security group ingress rule."
+  default     = [""]
+
```
Comment (bebosudo) [inline variables.tf:1680]:
```suggestion
```

**Review by bebosudo — APPROVED** [review]

**Review by bebosudo — APPROVED** [review]

File: examples/complete_public_ec2/main.tf [hunk 4]
```diff
@@ -0,0 +1,30 @@
+## -- Complete Example for Public EC2 instance -- ##
+module "terraform-seqera-module" {
```
Comment (pditommaso) [inline examples/complete_public_ec2/main.tf:2]:
```suggestion
module "terraform-seqera-aws" {
```

File: examples/complete_public_ec2/main.tf [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  # source  = "github.com/seqeralabs/terraform-seqera-module"
+  source = "/Users/adianny/seqera/git/terraform-seqera-module"
```
Comment (pditommaso) [inline examples/complete_public_ec2/main.tf:4]: This looks a left over of the development

**Review by pditommaso — CHANGES_REQUESTED** [review]

**Review by ashseqera — CHANGES_REQUESTED** [review]
@enekui  Noticed while adding EFS storage for the new ` monitoring `cluster that you have to add the `encrypted = true ` option while creating the EFS `resource "aws_efs_file_system" "eks_efs" ` otherwise the EFS data will be unencrypted

**Review by bebosudo — COMMENTED** [review]
Still some references to tower, there's a configmap called `kubernetes_config_map_v1.tower_app_configmap`

**Comment by bebosudo** [comment]
I'm not discussing the use of a configmap, but its name. Do we want to rename it to Seqera platform or something?

**Comment by pditommaso** [comment]
It's OK for now. There's a huge number of variable and settings prefixed with "tower". the idea is to keep this as near as possible to the current deployment

**Review by bebosudo — APPROVED** [review]
LGTM!

**Review by ashseqera — APPROVED** [review]
LGTM

**Comment by pditommaso** [comment]
Sure, can you please make me a summary of the new settings introduced by this PR and the default value?

**Comment by pditommaso** [comment]
Ok, thanks

**Pull request merged**